src/osd/PGLog.cc

```diff
@@ -711,18 +729,39 @@ void PGLog::_write_log_and_missing_wo_missing(
     }
   }
 
-  // process dirty_dups after log_keys_debug is filled, so dups do not
+  // process dups after log_keys_debug is filled, so dups do not
   // end up in that set
-  if (dirty_dups) {
+  if (dirty_to_dups != eversion_t()) {
     pg_log_dup_t min;
     t.omap_rmkeyrange(
       coll, log_oid,
       min.get_key_name(), log.dups.begin()->get_key_name());
```

Comment:
I don't quite follow why here we clear up to log.dups.begin(), but down below we clear from dirty_from_dups.  Seems like below we should also clear from logs.dups.rbegin()?  I gues it's moot since the keys will get overwritten if they're not deleted?

Reply:
the idea is to clear the dup_* keyspace from any extraneous stuff due to an import or bug, so we clear [min, dirty_to] and [dirty_from, max] - probably more of a likelihood with the regular pg log due to divergent entries, but it shouldn't hurt here - since these will only be used when merging logs, the perf impact should be minimal

Reply:
ah valgrind caught the problem though - that was meant to clearing to dirty_to_dups instead of log.dups.begin()